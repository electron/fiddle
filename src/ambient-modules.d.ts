declare module 'namor';
declare module 'shell-path';
