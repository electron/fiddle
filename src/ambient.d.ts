import * as MonacoType from 'monaco-editor';

import { App } from './renderer/app';

declare global {
  interface Window {
    ElectronFiddle: {
      app: App;
      appPaths: Record<string, string>;
      monaco: typeof MonacoType;
    };
  }
}
