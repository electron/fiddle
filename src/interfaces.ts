import { InstallState } from '@vertedinde/fiddle-core';

export type Files = Map<string, string>;

export type FileTransform = (files: Files) => Promise<Files>;

export enum VersionSource {
  remote = 'remote',
  local = 'local',
}

export enum GistActionType {
  publish = 'Publish',
  update = 'Update',
  delete = 'Delete',
}

export enum GistActionState {
  publishing = 'publishing',
  updating = 'updating',
  deleting = 'deleting',
  none = 'none',
}

export interface Version {
  version: string;
  name?: string;
  localPath?: string;
  node?: string;
}

export enum RunResult {
  SUCCESS = 'success', // exit code === 0
  FAILURE = 'failure', // ran, but exit code !== 0
  INVALID = 'invalid', // could not run
}

export interface RunnableVersion extends Version {
  state: InstallState;
  source: VersionSource;
  downloadProgress?: number;
}

export const enum ElectronReleaseChannel {
  stable = 'Stable',
  beta = 'Beta',
  nightly = 'Nightly',
}

export interface SetFiddleOptions {
  filePath?: string;
  templateName?: string;
  gistId?: string;
}

export interface SetUpMenuOptions {
  acceleratorsToBlock?: BlockableAccelerator[] | null;
  activeTemplate?: string | null;
}

export interface SetupRequest {
  fiddle?: SetFiddleOptions;
  version?: string;
  showChannels: ElectronReleaseChannel[];
  hideChannels: ElectronReleaseChannel[];
  useObsolete?: boolean;
}

export interface BisectRequest {
  setup: SetupRequest;
  goodVersion: string;
  badVersion: string;
}

export interface TestRequest {
  setup: SetupRequest;
}

export interface OutputEntry {
  text: string;
  timeString: string;
  isNotPre?: boolean;
}

export interface OutputOptions {
  bypassBuffer?: boolean;
  isNotPre?: boolean;
}

export interface GenericDialogOptions {
  type: GenericDialogType;
  ok: string;
  cancel?: string;
  wantsInput: boolean;
  defaultInput?: string;
  label: string | JSX.Element;
  placeholder?: string;
}

export interface Contributor {
  url: string;
  api: string;
  login: string;
  avatar: string;
  name: string;
  bio: string;
  location: string;
}

export interface Templates {
  [index: string]: string | Templates;
}

export const enum GenericDialogType {
  'confirm' = 'confirm',
  'warning' = 'warning',
  'success' = 'success',
}

export type EditorId = `${string}.${'js' | 'html' | 'css'}`;

export type EditorValues = Record<EditorId, string>;

// main.js gets special treatment: it is required as the entry point
// when we run fiddles or create a package.json to package fiddles.
export const MAIN_JS = 'main.js';

export const PACKAGE_NAME = 'package.json';

export type ArrowPosition = 'top' | 'left' | 'bottom' | 'right';

export const enum BlockableAccelerator {
  save = 'save',
  saveAs = 'saveAs',
}
