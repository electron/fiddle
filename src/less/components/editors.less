.mosaic-tile {
  animation: fadein 0.15s;
}

.mosaic-window-toolbar  {
  & > div {
    display: flex;
    justify-content: space-between;
    width: 100%;
    padding: 0 5px 0 5px;

    .mosaic-controls {
      transform: scale(0.75);
      transform-origin: right;
    }

    h5 {
      margin: 4px 0 0 0;
    }

    button {
      margin-left: 8px;
    }
  }
}

// TODO: support new file
// update if list of Editor ID changes
@editor-ids: main\.js, renderer\.js, index\.html, preload\.js, styles\.css;

// takes each editor ID and increase its z-index if the parent Mosaic root
// has focused__id class for that specific id.
each(@editor-ids, {
  .focused__@{value}.mosaic .mosaic-window.@{value} {
    z-index: 2;
  }
});

.mosaic-window {
  z-index: 1;
}

@keyframes fadein {
  from { opacity: 0; }
  to { opacity: 1; }
}
