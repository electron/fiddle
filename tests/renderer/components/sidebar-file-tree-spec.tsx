import * as React from 'react';

import { shallow } from 'enzyme';

import { EditorValues } from '../../../src/interfaces';
import { Editors } from '../../../src/renderer/components/editors';
import { SidebarFileTree } from '../../../src/renderer/components/sidebar-file-tree';
import {
  EditorMosaic,
  EditorPresence,
} from '../../../src/renderer/editor-mosaic';
import { createEditorValues } from '../../mocks/editor-values';
import { AppMock, StateMock } from '../../mocks/mocks';

describe('SidebarFileTree component', () => {
  let app: AppMock;
  let store: any;
  let editorMosaic: EditorMosaic;
  let editorValues: EditorValues;
  let stateMock: StateMock;

  beforeEach(() => {
    ({ app } = (window as any).ElectronFiddle);
    ({ state: stateMock } = app);
    store = {};
    editorValues = createEditorValues();
    editorMosaic = new EditorMosaic();
    editorMosaic.set(editorValues);
    store.editorMosaic = editorMosaic as any;
    stateMock.editorMosaic = editorMosaic as any;
  });

  it('renders', () => {
    const wrapper = shallow(<SidebarFileTree appState={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('reflects the visibility state of all icons', () => {
    editorMosaic.hide('index.html');
    const wrapper = shallow(<SidebarFileTree appState={store} />);

    // snapshot has an 'eye-off' icon
    expect(wrapper).toMatchSnapshot();
  });

  it('can bring up the Add File input', () => {
    const wrapper = shallow(<SidebarFileTree appState={store} />);
    const instance: any = wrapper.instance() as any;

    instance.setState({ action: 'add' });

    // snapshot has the input rendered
    expect(wrapper).toMatchSnapshot();
  });

  it('can toggle editor visibility', () => {
    const wrapper = shallow(<SidebarFileTree appState={store} />);
    const instance: any = wrapper.instance() as any;

    instance.toggleVisibility('index.html');

    expect(editorMosaic.files.get('index.html')).toBe(EditorPresence.Hidden);
  });

  it('can create new editors', () => {
    const wrapper = shallow(<SidebarFileTree appState={store} />);
    const instance: any = wrapper.instance() as any;

    expect(editorMosaic.files.get('tester.js')).toBe(undefined);
    instance.createEditor('tester.js');
    expect(editorMosaic.files.get('tester.js')).toBe(EditorPresence.Pending);
  });

  it('can delete editors', () => {
    const wrapper = shallow(<SidebarFileTree appState={store} />);
    const instance: any = wrapper.instance() as any;

    expect(editorMosaic.files.get('index.html')).toBe(EditorPresence.Pending);
    instance.removeEditor('index.html');
    expect(editorMosaic.files.get('index.html')).toBe(undefined);
  });

  it('can rename editors', async () => {
    const wrapper = shallow(<SidebarFileTree appState={store} />);
    const instance: any = wrapper.instance() as any;

    const EDITOR_NAME = 'index.html';
    const EDITOR_NEW_NAME = 'new_index.html';

    store.showInputDialog = jest.fn().mockResolvedValueOnce(EDITOR_NEW_NAME);

    expect(editorMosaic.files.get(EDITOR_NAME)).toBe(EditorPresence.Pending);

    await instance.renameEditor(EDITOR_NAME);

    expect(editorMosaic.files.get(EDITOR_NAME)).toBe(undefined);
    expect(editorMosaic.files.get(EDITOR_NEW_NAME)).toBe(
      EditorPresence.Pending,
    );
  });

  it('can reset the editor layout', () => {
    const wrapper = shallow(<SidebarFileTree appState={store} />);
    const instance: any = wrapper.instance() as any;

    editorMosaic.resetLayout = jest.fn();

    instance.resetLayout();

    expect(editorMosaic.resetLayout).toHaveBeenCalledTimes(1);
  });

  it('file is visible, click files tree, focus file content', async () => {
    const sidebarFileTree = shallow(<SidebarFileTree appState={store} />);
    const editors = shallow(<Editors appState={stateMock as any} />);
    const sidebarFileTreeInstance: any = sidebarFileTree.instance() as any;
    const editorsInstance: any = editors.instance() as any;

    sidebarFileTreeInstance.setFocusedFile('index.html');

    setTimeout(() => {
      expect(editorMosaic.files.get('index.html')).toBe(EditorPresence.Visible);
      expect(editorsInstance.state.focused).toBe('index.html');
    });
  });

  it('file is hidden, click files tree, make file visible and focus file content', function () {
    const sidebarFileTree = shallow(<SidebarFileTree appState={store} />);
    const editors = shallow(<Editors appState={stateMock as any} />);
    const sidebarFileTreeInstance: any = sidebarFileTree.instance() as any;
    const editorsInstance: any = editors.instance() as any;

    sidebarFileTreeInstance.toggleVisibility('index.html');

    expect(editorMosaic.files.get('index.html')).toBe(EditorPresence.Hidden);

    sidebarFileTreeInstance.setFocusedFile('index.html');

    setTimeout(() => {
      expect(editorMosaic.files.get('index.html')).toBe(EditorPresence.Visible);
      expect(editorsInstance.state.focused).toBe('index.html');
    });
  });
});
