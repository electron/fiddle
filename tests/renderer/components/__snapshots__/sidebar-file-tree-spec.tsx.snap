// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarFileTree component can bring up the Add File input 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <Blueprint3.Tree
    contents={
      Array [
        Object {
          "childNodes": Array [
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 0,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                index.html
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 1,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={true}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                main.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 2,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                preload.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 3,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                renderer.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 4,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                styles.css
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "className": "add-file-input",
              "icon": "document",
              "id": "add",
              "label": <input
                autoFocus={true}
                className="bp3-input bp3-fill bp3-small"
                id="new-file-input"
                onBlur={[Function]}
                onKeyDown={[Function]}
                style={
                  Object {
                    "padding": 0,
                    "width": "100%",
                  }
                }
              />,
            },
          ],
          "hasCaret": false,
          "icon": "folder-open",
          "id": "files",
          "isExpanded": true,
          "label": "Editors",
          "secondaryLabel": <Blueprint3.ButtonGroup
            minimal={true}
          >
            <Blueprint3.Tooltip2
              content="Add New File"
              hoverCloseDelay={0}
              hoverOpenDelay={1000}
              minimal={true}
              transitionDuration={100}
            >
              <Blueprint3.Button
                icon="add"
                onClick={[Function]}
                small={true}
              />
            </Blueprint3.Tooltip2>
            <Blueprint3.Tooltip2
              content="Reset Layout"
              hoverCloseDelay={0}
              hoverOpenDelay={1000}
              minimal={true}
              transitionDuration={100}
            >
              <Blueprint3.Button
                icon="grid-view"
                onClick={[Function]}
                small={true}
              />
            </Blueprint3.Tooltip2>
          </Blueprint3.ButtonGroup>,
        },
      ]
    }
  />
</div>
`;

exports[`SidebarFileTree component reflects the visibility state of all icons 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <Blueprint3.Tree
    contents={
      Array [
        Object {
          "childNodes": Array [
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 0,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                index.html
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-off"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 1,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={true}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                main.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 2,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                preload.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 3,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                renderer.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 4,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                styles.css
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
          ],
          "hasCaret": false,
          "icon": "folder-open",
          "id": "files",
          "isExpanded": true,
          "label": "Editors",
          "secondaryLabel": <Blueprint3.ButtonGroup
            minimal={true}
          >
            <Blueprint3.Tooltip2
              content="Add New File"
              hoverCloseDelay={0}
              hoverOpenDelay={1000}
              minimal={true}
              transitionDuration={100}
            >
              <Blueprint3.Button
                icon="add"
                onClick={[Function]}
                small={true}
              />
            </Blueprint3.Tooltip2>
            <Blueprint3.Tooltip2
              content="Reset Layout"
              hoverCloseDelay={0}
              hoverOpenDelay={1000}
              minimal={true}
              transitionDuration={100}
            >
              <Blueprint3.Button
                icon="grid-view"
                onClick={[Function]}
                small={true}
              />
            </Blueprint3.Tooltip2>
          </Blueprint3.ButtonGroup>,
        },
      ]
    }
  />
</div>
`;

exports[`SidebarFileTree component renders 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <Blueprint3.Tree
    contents={
      Array [
        Object {
          "childNodes": Array [
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 0,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                index.html
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 1,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={true}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                main.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 2,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                preload.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 3,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                renderer.js
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
            Object {
              "hasCaret": false,
              "icon": "document",
              "id": 4,
              "isSelected": false,
              "label": <Blueprint.ContextMenu2
                className="pointer"
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="redo"
                      intent="primary"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Rename"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      icon="remove"
                      intent="danger"
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Delete"
                    />
                  </Blueprint3.Menu>
                }
                onClick={[Function]}
              >
                styles.css
              </Blueprint.ContextMenu2>,
              "secondaryLabel": <Blueprint3.ButtonGroup>
                <Blueprint3.Tooltip2
                  content="Toggle Visibility"
                  hoverCloseDelay={0}
                  hoverOpenDelay={1000}
                  minimal={true}
                  transitionDuration={100}
                >
                  <Blueprint3.Button
                    minimal={true}
                    onClick={[Function]}
                  >
                    <Blueprint3.Icon
                      icon="eye-open"
                    />
                  </Blueprint3.Button>
                </Blueprint3.Tooltip2>
              </Blueprint3.ButtonGroup>,
            },
          ],
          "hasCaret": false,
          "icon": "folder-open",
          "id": "files",
          "isExpanded": true,
          "label": "Editors",
          "secondaryLabel": <Blueprint3.ButtonGroup
            minimal={true}
          >
            <Blueprint3.Tooltip2
              content="Add New File"
              hoverCloseDelay={0}
              hoverOpenDelay={1000}
              minimal={true}
              transitionDuration={100}
            >
              <Blueprint3.Button
                icon="add"
                onClick={[Function]}
                small={true}
              />
            </Blueprint3.Tooltip2>
            <Blueprint3.Tooltip2
              content="Reset Layout"
              hoverCloseDelay={0}
              hoverOpenDelay={1000}
              minimal={true}
              transitionDuration={100}
            >
              <Blueprint3.Button
                icon="grid-view"
                onClick={[Function]}
                small={true}
              />
            </Blueprint3.Tooltip2>
          </Blueprint3.ButtonGroup>,
        },
      ]
    }
  />
</div>
`;
